### PR TITLE
feat: add repository insight view pages to workspaces

### PR DESCRIPTION
**Pull request by brandonroberts**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Mobile & Desktop Screenshots/Recordings

image

image

image

image

## Steps to QA

1. Navigate to deploy preview
2. Login
3. Go to workspaces home
4. Click on Repository Insights from the left menu
5. Navigate around through a repository insight

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by bdougie — APPROVED** [review]
This looks great. Mention things I see that are unrelated. I realize this may get covered next week and documented in issues too. 

- When I collapse the side panel and click a new tab; the side panel reopens - https://github.com/open-sauced/app/issues/2688
- When I log in I land on the  /feed with a ?redirected param. I try to click the /workspaces link in the nav and nothing happens. https://github.com/open-sauced/app/issues/2691

I'll find issues if they exist, or create new ones.

**Comment by jpmcb** [comment]
cc @bdougie I noticed the same with the `/workspaces` top header link not working: https://github.com/open-sauced/app/pull/2689#pullrequestreview-1885758519 must have snuck in somewhere although it works fine on `beta`, maybe it's a deploy preview environment problem?

**Comment by bdougie** [comment]
> cc @bdougie I noticed the same with the `/workspaces` top header link not working: [#2689 (review)](https://github.com/open-sauced/app/pull/2689#pullrequestreview-1885758519) must have snuck in somewhere although it works fine on `beta`, maybe it's a deploy preview environment problem?

I just test it again and narrowed it down to the `?redirectedFrom=%2F` param check preventing navigation.

**Comment by brandonroberts** [comment]
> > cc @bdougie I noticed the same with the `/workspaces` top header link not working: [#2689 (review)](https://github.com/open-sauced/app/pull/2689#pullrequestreview-1885758519) must have snuck in somewhere although it works fine on `beta`, maybe it's a deploy preview environment problem?
> 
> I just test it again and narrowed it down to the `?redirectedFrom=%2F` param check preventing navigation. Something to do with the state that causes this edge case.

I'll take a look at that next

File: pages/workspaces/[workspaceId]/repository-insights/[insightId]/activity.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      <>
+        <SEO
+          title={`${insight.name} | Open Sauced Insights `}
+          description={`${insight.name} Insights on OpenSauced`}
+          image={ogImage}
+          twitterCard="summary_large_image"
+        />
+      </>
+    );
+  }
```
Comment (nickytonline) [inline pages/workspaces/[workspaceId]/repository-insights/[insightId]/activity.tsx:40]: refactor (blocking): use the `useHasMounted()` hook from `import { useHasMounted } from "lib/hooks/useHasMounted";` Same for other pages/components in the PR.

```suggestion
  const hasMounted = useHasMounted(); 
  
  if (!hasMounted) {
    return (
        <SEO
          title={`${insight.name} | Open Sauced Insights `}
          description={`${insight.name} Insights on OpenSauced`}
          image={ogImage}
          twitterCard="summary_large_image"
        />
    );
  }
```

**Review by nickytonline — CHANGES_REQUESTED** [review]
This is looking great @brandonroberts! A few change requests.

File: components/organisms/ToolList/nav.tsx [hunk 2]
```diff
@@ -17,11 +17,19 @@ interface NavProps {
   filterName?: string | string[] | undefined;
   selectedFilter?: string | string[] | undefined;
   username?: string | string[] | undefined;
+  workspaceId?: string;
```
[the remark this answers is not in the document]
Comment (brandonroberts) [reply components/organisms/ToolList/nav.tsx:20]: Yes, that's correct. We still have the other pages at the moment

File: pages/workspaces/[workspaceId]/repository-insights/[insightId]/activity.tsx (hunk 1 above)
Comment (brandonroberts) [reply pages/workspaces/[workspaceId]/repository-insights/[insightId]/activity.tsx:40]: Ok cool. Wasn't aware of this hook

**Review by nickytonline — APPROVED** [review]
:shipit:
